src/pages/header.js

```diff
@@ -0,0 +1,94 @@
+import * as React from "react"
+import { Link, graphql } from "gatsby"
+
+import styled from "styled-components"
+import { Composition, Box } from "atomic-layout"
+
+const StyledLogo = styled.div`
+  padding: var(--spacing-12);
+`
+
+const StyledUl = styled.ul`
+  list-style: none;
+  display: flex;
+  justify-content: space-around;
+  margin: 0;
+`
+const StyledLi = styled.li`
+  margin: 0;
+
+  a {
+    display: block;
+    text-transform: uppercase;
+    font-size: 14px;
+    letter-spacing: 2px;
+
+    padding: var(--spacing-4);
+    background-image: linear-gradient(125deg, black 50%, red 50% 100%);
+    -webkit-background-clip: text;
+    -webkit-text-fill-color: transparent;
+    background-size: 200%;
+    background-position: 0%;
+    transition: background-position 0.8s;
+
+    :hover {
+      background-position: 110%;
+    }
+  }
+`
+
+const Header = ({ data }) => {
```

Comment:
I recommend refactoring this a little bit:

1. Move this `Header` component to a new file at `src/components/Header.jsx`.
2. Create a new component called `DefaultLayout` in `src/components`. Structure it like this:

```jsx
// Your newly created "Header":
import { Header } from './Header'

export function DefaultLayout({ children }) {
  return (
    <>
      <Header />
      <main>{children}</main>
      {/* In the future you'll have Footer here as well */}
    </>
  )
}
```

3. Then, finally, in your `src/pages/header.js` use the `DefaultLayout`:
```diff
-const Header = ({ data }) => {
+const Page = ({ data }) => {
  return (
+    <DefaultLayout>
        {posts.map(/* what you already have now */)}
+    </DefaultLayout>
  )
}

-export default Header
+export default Page
```